index.js

```diff
@@ -19,10 +28,76 @@ module.exports = {
         return converter.convert(data, options, cb);
       });
     }
-    return cb(null, {
-      result: false,
-      reason: 'input type:' + input.type + ' is not valid'
-    });
+    else if (input.type === 'folder') {
+      let filesPathArray = input.data,
+        convertedSpecs = [],
+        rootFiles = parse.getRootFiles(filesPathArray);
+
+      async.eachSeries(rootFiles, (rootFile, callback) => {
+        parse
+        // will merge all the files in the folder
+          .mergeFiles(rootFile, OasResolverOptions)
+          .then((spec) => {
+            converter.convert(spec, options, (err, result) => {
+              if (err) {
+                return callback(null, {
+                  result: false,
+                  reason: err
+                });
+              }
+              // eslint-disable-next-line no-else-return
+              else {
+                convertedSpecs.push(result);
+                return callback(null);
+              }
+
+            });
+          })
+          .catch((err) => {
+            return callback(null, {
+              result: false,
+              reason: err
+            });
+          });
+      }, () => {
+
+        var conversionResult = false,
+          convertedCollections = [],
+          reasonForFail;
+
+        _.forEach(convertedSpecs, (convertedSpec) => {
+          if (convertedSpec.result) {
+            conversionResult = convertedSpec.result;
+            convertedCollections.push(convertedSpec.output[0]);
```

Comment:
Why are we restricting the push to output[0]? We'll start returning environments - this code should ideally not need to be changed.  The return value comes from

```
return callback(null, {
        result: true,
        output: [{
          type: 'collection',
          data: this.generatedStore.collection.toJSON()
        }]
      });

```

At this point, you should ideally be filtering for output.type=collection, and then adding to convertedCollections